cmd/client_test.go

```diff
@@ -111,3 +111,59 @@ func TestClient_CreateJobSpec(t *testing.T) {
 		assert.Equal(t, test.nJobs, len(numberOfJobs))
 	}
 }
+
+func TestClient_CreateJobRun(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+	client, _ := cltest.NewClientAndRenderer(app.Store.Config)
+
+	tests := []struct {
+		name    string
+		json    string
+		jobType int
+		badID   bool
+		errored bool
+	}{
+		{"Create_Success", `{"value": 100}`, 0, false, false},
+		{"EmptyBody", ``, 0, false, false},
+		{"InvalidBody", `{`, 0, false, true},
+		{"WithoutWebInitiator", ``, 0, true, true},
+		{"NotFound", ``, 1, false, true},
+	}
+
+	for _, tt := range tests {
+		test := tt
+		t.Run(test.name, func(t *testing.T) {
+			var jobSpec models.JobSpec
+			var args []string
+			set := flag.NewFlagSet("run", 0)
+
+			switch test.jobType {
+			case 0:
+				jobSpec = cltest.NewJobWithWebInitiator()
+			case 1:
+				jobSpec = cltest.NewJobWithLogInitiator()
+			}
+			assert.Nil(t, app.Store.SaveJob(&jobSpec))
+
+			if test.badID {
+				args = append(args, "badID")
+			} else {
+				args = append(args, jobSpec.ID)
+			}
+
+			if len(test.json) > 0 {
+				args = append(args, test.json)
```

Comment:
Does it actually make a difference if you append an empty string?

Reply:
if you append empty string its an invalid filepath it checks for either valid json/filepath. So it will error out if you pass a second but empty argument.